src/scripts/createAsar.ts

```diff
@@ -0,0 +1,48 @@
+
+import * as path from 'path';
+
+import {getProjectConfig} from '../utils/getProjectConfig';
+import {getRootDirectory} from '../utils/getRootDirectory';
+
+const asar = require('asar');
+const fs = require('fs-extra');
+const glob = require('glob');
+const openfinSign = require('openfin-sign');
+
+/**
+ * Creates a zip archive of the service provider.
+ */
+export async function createAsar() {
+    const {SERVICE_NAME} = getProjectConfig();
+    const output = path.resolve(getRootDirectory(), 'dist', 'asar');
+
+    fs.emptyDirSync(output);
+
+    [
+        path.resolve(getRootDirectory(), 'dist', 'provider'),
+        path.resolve(getRootDirectory(), 'res', 'provider')
+    ].forEach((file) => {
+        fs.copySync(file, output);
+    });
+
+    [
+        path.resolve(getRootDirectory(), 'dist', 'client', `openfin-${SERVICE_NAME}.js`)
+    ].forEach((file) => {
+        fs.copyFileSync(file, path.resolve(output, path.basename(file)));
+    });
```

Comment:
Does the `asar` lib support streams? Would ideally create the archive without local file copying, as per `createProviderZip.ts`.

Reply:
asar does not support streams.  Also,  each file needs to be signed, in addition to fdc3.asar.